front/src/ctfnote/auth.ts

```diff
@@ -120,3 +121,67 @@ export function useLogout() {
     document.location.reload();
   };
 }
+
+// LDAP Login GraphQL mutation (will be generated after server is running)
+const LOGIN_WITH_LDAP_MUTATION = gql`
+  mutation AuthenticateWithLdap($username: String!, $password: String!) {
+    authenticateWithLdap(username: $username, password: $password) {
+      jwt
+    }
+  }
+`;
+
+export function useLdapLogin() {
```

Comment:
After moving the graphql to the graphql file, you can just use similar code as `useLogin`.